src/VisualStudio/TestUtilities2/ProjectSystemShim/Framework/MockVsFileChangeEx.vb

```diff
@@ -14,8 +15,8 @@ Namespace Microsoft.VisualStudio.LanguageServices.UnitTests.ProjectSystemShim.Fr
         Implements IVsAsyncFileChangeEx
 
         Private ReadOnly _lock As New Object
-        Private ReadOnly _watchedFiles As New List(Of WatchedEntity)
-        Private ReadOnly _watchedDirectories As New List(Of WatchedEntity)
+        Private _watchedFiles As ImmutableList(Of WatchedEntity) = ImmutableList(Of WatchedEntity).Empty
```

Comment:
I'm not entirely sure why these were changed. Is it a perf concern? List actually seems better since we treat them as mutable under a lock.

Reply:
To properly test this fix, I need to implement support in the mock to test the sequence of events of:

1. Capture the list of watched entities.
2. Unsubscribe from something.
3. Raise the event from the captured list in 1.

Making this immutable means I can easily capture it for later.